pkg/env/armhelper.go

```diff
@@ -113,23 +121,26 @@ func newARMHelper(ctx context.Context, log *logrus.Entry, env Interface) (ARMHel
 		log: log,
 		env: env,
 
+		fpGraphClient:   fpGraphClient,
 		roleassignments: authorization.NewRoleAssignmentsClient(env.Environment(), env.SubscriptionID(), armAuthorizer),
-		applications:    graphrbac.NewApplicationsClient(env.Environment(), env.TenantID(), fpGraphAuthorizer),
 	}, nil
 }
 
 func (ah *armHelper) EnsureARMResourceGroupRoleAssignment(ctx context.Context, fpAuthorizer autorest.Authorizer, resourceGroup string) error {
 	ah.log.Print("ensuring resource group role assignment")
 
-	res, err := ah.applications.GetServicePrincipalsIDByAppID(ctx, ah.env.FPClientID())
+	principalID, err := utilgraph.GetServicePrincipalIDByAppID(ctx, ah.fpGraphClient, ah.env.FPClientID())
 	if err != nil {
 		return err
 	}
+	if principalID == nil {
+		return fmt.Errorf("appId '%s' not found", ah.env.FPClientID())
```

Comment:
Confused on this result - wouldn't this be `"service principal for appId '%s' was not found"`?

Reply:
I think it could be either case, now that I think about it.  The way the function does the principal ID query doesn't really allow us to distinguish whether the appId itself was bad or there's just no service principals for it.

Do you think it matters enough to rewrite the query?  I think we'd have to do two graph calls then instead of one (one to validate the appId, the other to search for principals).

Reply:
I'll change the error message to:
```
No service principals found for application ID '%s'
```
I think that covers both possibilities.  Good call out.